### PR TITLE
Dot support

### PR DESCRIPTION
**Pull request by alemagio**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Can you add `out` to `.gitignore`?

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -41,6 +45,8 @@ function fastifyView (fastify, opts, next) {
     return
   }
 
+  const dotRender = type === 'dot' ? viewDot(preProcessDot(templatesDir, options)) : null
```
Comment (Eomm) [inline index.js:48]:
```suggestion
  const dotRender = type === 'dot' ? viewDot.call(fastify, preProcessDot(templatesDir, options)) : null
```

File: index.js [hunk 2]
```diff
@@ -192,6 +199,34 @@ function fastifyView (fastify, opts, next) {
...
+        destination: destinationDir
+      }
+    ))
+
+    // .jst files are compiled to .js files so we need to require them
+    for (const file of readdirSync(destinationDir, { withFileTypes: false })) {
+      renderer[basename(file, '.js')] = require(resolve(join(destinationDir, file)))
+    }
+    if (Object.keys(renderer).length === 0) {
+      console.warn(`WARN: no template found in ${templatesDir}`)
```
Comment (Eomm) [inline index.js:224]:
```suggestion
      this.log.warn(`WARN: no template found in ${templatesDir}`)
```

**Review by Eomm — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**